### PR TITLE
Manual cherry pick for #65627

### PR DESCRIPTION
**Pull request by draganescu**
This is a manual cherry pick PR for #65627.

### CODE REVIEW

File: packages/block-editor/src/components/block-popover/index.js [hunk 1]
```diff
@@ -74,12 +77,38 @@ function BlockPopover(
 		};
 	}, [ selectedElement ] );
 
+	const { isZoomOut, parentSectionBlock, isSectionSelected } = useSelect(
+		( select ) => {
+			const {
+				isZoomOut: isZoomOutSelector,
+				getSectionRootClientId,
+				getParentSectionBlock,
```
Comment (getdave) [inline packages/block-editor/src/components/block-popover/index.js:85]: I think we're missing whatever PR included `getParentSectionBlock`.

```
index.js:92 Uncaught TypeError: getParentSectionBlock is not a function
    at index.js:92:6
    at index.js:154:11
    at __unstableMarkListeningStores (registry.js:123:20)
    at Object.__unstableMarkListeningStores (registry.js:204:30)
    at updateValue (index.js:153:31)
    at index.js:203:3
    at useMappingSelect (index.js:228:34)
    at useSelect (index.js:317:5)
    at BlockPopover (index.js:80:72)
    at renderWithHooks (react-dom.js?ver=18:15496:20)
```

**Review by getdave — COMMENTED** [review]
Thanks for following up. This errors when activating Zoom Out. I think we're missing a PR that didn't get backported.

File: packages/block-editor/src/components/block-popover/index.js (hunk 1 above)
Comment (draganescu) [reply packages/block-editor/src/components/block-popover/index.js:85]: ups! on it

**Comment by draganescu** [comment]
@getdave I've brought over the missing selectors. It does not crash anymore and the build seems clean.

**Comment by kevin940726** [comment]
@draganescu Thank you! Which PR was missing and should we just cherry-pick that too instead of patching it here? If it's a rabbit hole and can't be cherry-picked cleanly, do you think we should squash the commits (in this PR) so that when we "rebase and merge" this it can maintain a clean commit history?

**Comment by draganescu** [comment]
@kevin940726 the PR to cherry-pick is [Select Mode: Use the content-only behavior in select mode](https://github.com/WordPress/gutenberg/pull/65204#top) ... not sure what to do with it.

**Review by ajlende — COMMENTED** [review]
I don't think we want all of https://github.com/WordPress/gutenberg/pull/65204 backported. It's categorized as https://github.com/WordPress/gutenberg/labels/%5BType%5D%20Enhancement and it modifies 46 files.

What @draganescu copied over here is a much more minimal set of changes for the bug fix. The new selectors are private, so that shouldn't cause any issues.

**Review by ajlende — APPROVED** [review]

**Comment by SantosGuillamot** [comment]
I think the failing performance tests should be fixed after [this pull request](https://github.com/WordPress/gutenberg/pull/66196). It might require a rebase.

**Comment by cbravobernal** [comment]
I just did the rebase. Will merge once tests pass.

**Pull request merged**